src/test/java/ua/tc/marketplace/service/impl/CategoryServiceImplTest.java

```diff
@@ -0,0 +1,155 @@
+import ua.tc.marketplace.service.impl.CategoryServiceImpl;
+import ua.tc.marketplace.util.mapper.CategoryMapper;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@ExtendWith(MockitoExtension.class)
+class CategoryServiceImplTest {
+    @Mock
+    private ClassificationAttributeRepository attributeRepository;
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @Test
+    void findCategoryById_shouldFindCategory_whenExists() {
+        Long categoryId = 1L;
+        Category category = new Category(); // create a sample Category entity
+        CategoryDTO categoryDto = new CategoryDTO(); // create a sample CategoryDTO
+
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.of(category));
+        when(categoryMapper.toCategoryDto(category)).thenReturn(categoryDto);
+
+        CategoryDTO result = categoryService.findById(categoryId);
+
+        assertEquals(categoryDto, result);
+        verify(categoryRepository, times(1)).findById(categoryId);
+        verify(categoryMapper, times(1)).toCategoryDto(category);
+    }
+
+    @Test
+    void findCategoryById_shouldThrow_whenNotExists() {
+        Long categoryId = 1L;
+
+        when(categoryRepository.findById(categoryId)).thenReturn(Optional.empty());
+
+        assertThrows(CategoryNotFoundException.class, () -> categoryService.findById(categoryId));
+
+        verify(categoryRepository, times(1)).findById(categoryId);
+        verify(categoryMapper, never()).toCategoryDto(any(Category.class));
+    }
+
+    @Test
+    void createCategory_shouldCreate_whenValidInput() {
+        CreateCategoryDTO createCategoryDto = new CreateCategoryDTO(); // create a sample CreateCategoryDTO
+        Category category = new Category(); // create a sample Category entity
+        CategoryDTO createdCategoryDto = new CategoryDTO(); // create a sample created CategoryDTO
+        List<ClassificationAttribute> attributes = Collections.singletonList(new ClassificationAttribute()); // create a list of attributes
+
+        when(categoryMapper.toEntity(createCategoryDto)).thenReturn(category);
+        when(categoryRepository.save(category)).thenReturn(category);
+        when(categoryMapper.toCategoryDto(category)).thenReturn(createdCategoryDto);
+
+        // Mock the attribute repository call
+        when(attributeRepository.findAllById(createCategoryDto.getAttributeIds())).thenReturn(attributes);
+
+        CategoryDTO result = categoryService.createCategory(createCategoryDto);
+
+        assertEquals(createdCategoryDto, result);
+        verify(categoryRepository, times(1)).save(category);
+        verify(categoryMapper, times(1)).toEntity(createCategoryDto);
+        verify(categoryMapper, times(1)).toCategoryDto(category);
+    }
+
+    @Test
+    void updateCategory_shouldUpdate_whenValidInput() {
```

Comment:
this test fails `java.lang.NullPointerException: Cannot invoke "java.util.List.stream()" because "attributeIds" is null`